### PR TITLE
Attempt to clean up the upload file if it was duplicated/renamed during deployment

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-61082]

This is a workaround which probably requires a better solution.

Related to https://github.com/OctopusDeploy/Issues/issues/7382

Related to https://github.com/OctopusDeploy/Calamari/pull/1155

### CODE REVIEW

File: source/Calamari.AzureAppService/Behaviors/AzureAppServiceZipDeployBehaviour.cs [hunk 1]
```diff
@@ -126,33 +126,53 @@ public async Task Execute(RunningDeployment context)
...
-            Log.Verbose($"Retrieving publishing profile for App Service to determine correct deployment endpoint.");
-            using var publishingProfileXmlStream = await armClient.GetPublishingProfileXmlWithSecrets(targetSite);
-            var publishingProfile = await PublishingProfile.ParseXml(publishingProfileXmlStream);
-
-            Log.Verbose($"Using deployment endpoint '{publishingProfile.PublishUrl}' from publishing profile.");
-
-            Log.Info($"Uploading package to {targetSite.SiteAndSlot}");
-
-            if (packageProvider.SupportsAsynchronousDeployment && FeatureToggle.AsynchronousAzureZipDeployFeatureToggle.IsEnabled(context.Variables))
+            bool uploadFileNeedsCleaning = false;
```
Comment (IsaacCalligeros95) [inline source/Calamari.AzureAppService/Behaviors/AzureAppServiceZipDeployBehaviour.cs:129]: Will this be required in the LegacyAzureAppServiceBehaviour as well?

File: source/Calamari.AzureAppService/Behaviors/AzureAppServiceBehaviour.cs [hunk 2]
```diff
@@ -1 +0,0 @@
-﻿
```
Comment (IsaacCalligeros95) [inline source/Calamari.AzureAppService/Behaviors/AzureAppServiceBehaviour.cs:1]: Thanks for the cleanup!

**Review by IsaacCalligeros95 — CHANGES_REQUESTED** [review]
I think the LegacyAzureAppServiceBehaviour will require the change aswell.

File: source/Calamari.AzureAppService/Behaviors/AzureAppServiceZipDeployBehaviour.cs (hunk 1 above)
Comment (benPearce1) [reply source/Calamari.AzureAppService/Behaviors/AzureAppServiceZipDeployBehaviour.cs:129]: good question, let me check

Comment (benPearce1) [reply source/Calamari.AzureAppService/Behaviors/AzureAppServiceZipDeployBehaviour.cs:129]: yes, missed that, thanks for the catch

**Review by IsaacCalligeros95 — APPROVED** [review]
LGTM

**Pull request merged**